src/generator/AutoRest.CSharp/Templates/Rest/Client/MethodTemplate.cshtml

```diff
@@ -479,7 +479,7 @@ else
         }
         else
         {
-        @:_result.Body = Microsoft.Rest.Serialization.SafeJsonConvert.DeserializeObject<@(Model.DefaultResponse.Body.AsNullableType())>(_defaultResponseContent, @(Model.GetDeserializationSettingsReference(Model.DefaultResponse.Body)));
+        @:_result.Body = Microsoft.Rest.Serialization.SafeJsonConvert.DeserializeObject<@(Model.DefaultResponse.Body.AsNullableType(Model. IsXNullableReturnType))>(_defaultResponseContent, @(Model.GetDeserializationSettingsReference(Model.DefaultResponse.Body)));
```

Comment:
valid point, but does `Model.IsXNullableReturnType` reflect whether the default response is supposed to be nullable? thought it's about the succeeding responses.